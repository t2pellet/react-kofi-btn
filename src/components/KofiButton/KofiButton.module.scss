.KofiBtnContainer {
    display:inline-block;
    white-space:nowrap;
    min-width: 150px;

    .KofiButton {
        display: flex;
        flex-direction: row;
        justify-content: center;
        align-items: center;
        height: 100%;
        gap: 0.25rem;
        text-decoration: none;
        padding-left: 1rem;
        padding-right: 1rem;
        box-shadow: 1px 1px 0 rgba(0, 0, 0, 0.2);
        overflow-wrap:break-word;

        &:hover {
            opacity:.85;
        }

        .KofiText {
            color: #fff;
            text-align: center;
            font-weight: 700;
            line-height: 14px;
            height: 14px;
            font-size: 14px;
            letter-spacing: -0.15px;
            font-family:'Quicksand',Helvetica,Century Gothic,sans-serif !important;
            text-shadow: 0 1px 1px rgba(34, 34, 34, 0.05);
        }

        .KofiImg {
            height: 13px;
            width:20px;
            border:none;
            content:url('https://storage.ko-fi.com/cdn/cup-border.png');
            animation: kofi-wiggle 3s infinite;
        }
    }
}

@keyframes kofi-wiggle {
    0%{
        transform:rotate(0) scale(1)
    }
    60%{
        transform:rotate(0) scale(1)
    }
    75%{
        transform:rotate(0) scale(1.12)
    }
    80%{
        transform:rotate(0) scale(1.1)
    }
    84%{
        transform:rotate(-10deg) scale(1.1)
    }
    88%{
        transform:rotate(10deg) scale(1.1)
    }
    92%{
        transform:rotate(-10deg) scale(1.1)
    }
    96%{
        transform:rotate(10deg) scale(1.1)
    }
    100%{
        transform:rotate(0) scale(1)
    }
}