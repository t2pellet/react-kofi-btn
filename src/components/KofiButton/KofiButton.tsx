import React, { useCallback, useEffect, useMemo } from 'react'
import WebFont from 'webfontloader'
import styles from './KofiButton.module.scss'

const buttonSizings = {
    sm: {
        width: 150,
        height: 33,
        fontSize: 14,
        iconHeight: 13,
        iconWidth: 20,
        border: 7,
        gap: 8,
    },
    md: {
        width: 200,
        height: 44,
        fontSize: 16,
        iconHeight: 15,
        iconWidth: 23,
        border: 12,
        gap: 12,
    },
    lg: {
        width: 300,
        height: 66,
        fontSize: 20,
        iconHeight: 19,
        iconWidth: 29,
        border: 28,
        gap: 16,
    },
}

export type KofiButtonProps = {
    size?: keyof typeof buttonSizings
    color?: string
    label: string
    user: string
    onClick?: (user: string) => void
}

const KofiButton: React.FC<KofiButtonProps> = ({
    size = 'sm',
    color = '#29abe0',
    label,
    user,
    onClick,
}) => {
    useEffect(() => {
        WebFont.load({
            google: {
                families: ['Quicksand'],
            },
        })
    }, [])

    const buttonSizing = useMemo(() => buttonSizings[size], [size])
    const openLink = useCallback(
        (user: string) => {
            if (onClick !== undefined) {
                onClick(user)
            } else if (window) {
                window.open(`https://ko-fi.com/${user}`, '_blank')
            }
        },
        [onClick],
    )

    return (
        <div
            className={styles.KofiBtnContainer}
            style={{ width: buttonSizing.width, height: buttonSizing.height }}
        >
            <a
                title='Support me on ko-fi.com'
                className={styles.KofiButton}
                style={{
                    backgroundColor: color,
                    borderRadius: buttonSizing.border,
                    gap: buttonSizing.gap,
                }}
                onClick={() => openLink(user)}
            >
                <img
                    src='https://storage.ko-fi.com/cdn/cup-border.png'
                    alt='Ko-fi donations'
                    width={buttonSizing.iconWidth}
                    height={buttonSizing.iconHeight}
                    className={styles.KofiImg}
                />
                <span
                    className={styles.KofiText}
                    style={{
                        lineHeight: `${buttonSizing.fontSize}px`,
                        height: buttonSizing.fontSize,
                        fontSize: buttonSizing.fontSize,
                    }}
                >
                    {label}
                </span>
            </a>
        </div>
    )
}

export default KofiButton
