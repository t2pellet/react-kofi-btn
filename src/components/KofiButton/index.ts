import KofiButton from './KofiButton'
import { KofiButtonProps } from './KofiButton'

export type { KofiButtonProps }
export default KofiButton
