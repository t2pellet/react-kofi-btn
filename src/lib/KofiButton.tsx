import React from 'react'
import './KofiButton.css'

type KofiButtonProps = {
    color?: string
    label: string
    user: string
}

const KofiButton: React.FC<KofiButtonProps> = ({
    color = '#29abe0',
    label,
    user,
}) => {
    return (
        <div className='KofiBtnContainer'>
            <a
                title='Support me on ko-fi.com'
                className='KofiButton'
                style={{ backgroundColor: color }}
                href={`https://ko-fi.com/${user}`}
                target='_blank'
                rel='noreferrer'
            >
                <img
                    src='https://storage.ko-fi.com/cdn/cup-border.png'
                    alt='Ko-fi donations'
                    className='KofiImg'
                />
                <span className='KofiText'>{label}</span>
            </a>
        </div>
    )
}

export default KofiButton
