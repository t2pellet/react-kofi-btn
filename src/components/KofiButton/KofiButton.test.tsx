import React from 'react'
import KofiButton from '.'
import { render } from '@testing-library/react'

describe('KofiButton', () => {
    it('renders properly', () => {
        const { container } = render(
            <KofiButton label='Support me' user='t2pellet' />,
        )
        expect(container).toMatchSnapshot()
    })
})
