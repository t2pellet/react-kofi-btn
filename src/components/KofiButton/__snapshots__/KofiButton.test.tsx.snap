// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KofiButton renders properly 1`] = `
<div>
  <div
    class="KofiBtnContainer"
  >
    <a
      class="KofiButton"
      href="https://ko-fi.com/t2pellet"
      rel="noreferrer"
      style="background-color: rgb(41, 171, 224);"
      target="_blank"
      title="Support me on ko-fi.com"
    >
      <img
        alt="Ko-fi donations"
        class="KofiImg"
        src="https://storage.ko-fi.com/cdn/cup-border.png"
      />
      <span
        class="KofiText"
      >
        Support me
      </span>
    </a>
  </div>
</div>
`;
